python/cudf/cudf/_libxx/parquet.pyx

```diff
@@ -227,4 +231,40 @@ cpdef write_parquet(
                                   tbl_meta.get(),
                                   comp_type,
                                   stat_freq)
-        parquet_writer(args)
+
+    if metadata_file_path is not None:
+        args.metadata_out_file_path = str.encode(metadata_file_path)
+        args.return_filemetadata = True
+
+    with nogil:
+        out_metadata_c = move(parquet_writer(args))
+
+    if metadata_file_path is not None:
+        out_metadata_py = \
+            out_metadata_c.get().data()[:out_metadata_c.get().size()]
```

Comment:
Why do you need to slice this? Cython should be able to implicitly convert a `vector[uint8_t]` to a Python list to return.

Reply:
Not sure, it was suggested to me on slack -> can I just do out_metadata_py = out_metadata_c ?

Reply:
Either way, this will give us a Python List of PyInts which is definitely not what we're looking for from here. From my perspective we have two options:
1) Create an opaque Cython class that holds a reference to the unique_ptr as an attribute. Can optionally give it buffer protocol later if desired to allow it to be used from Python land as if it's an array of bytes.
2) Return a memoryview, but this would involve releasing the unique_ptr and managing the memory lifetime via Cython / Python which is non-ideal to say the least. It's possible via Cython arrays: https://cython.readthedocs.io/en/latest/src/userguide/memoryviews.html#cython-arrays but I'd prefer Option 1.

Reply:
I'm afraid that goes beyond my current cython capabilities -> do we have an example in cudf of a class wrapper over a cpp object ? I think we'll need the buffer protocol support right away, since dask will need to not only be able to read this array to either pass it around or write it to disk, but also will need to create such an array (going the other way) to pass in to the merge function.

Reply:
How about a python bytearray using PyBytes_From_StringAndSize ? (that looked reasonably simple from a bit of googling around)

Reply:
>> How about a python bytearray using PyBytes_From_StringAndSize ? (that looked reasonably simple from a bit of googling around)

> "Return a new bytes object with a **copy** of the string v as value and length len on success"

It will copy construct it with a new memory allocation which is expensive.

Someone from my team can jump in and help with the Cython part.

Reply:
cc @brandon-b-miller can you tackle / help out with this?